core/src/avm2/optimize.rs

```diff
@@ -11,48 +11,80 @@ use std::collections::HashSet;
 use swf::avm2::types::Index;
 
 #[derive(Clone, Copy, Debug)]
-enum ValueType<'gc> {
-    // Either a class, or null.
-    Class(ClassObject<'gc>),
-    Int,
-    Uint,
-    Number,
-    Boolean,
-    Null,
-    Any,
+struct Value<'gc> {
```

Comment:
We should probably name this something else to avoid conflicting with AVM2 `Value`

Reply:
I wanted `Value` exactly because it only makes sense in optimizer's context ;D
And I wanted to keep it short, to reduce boilerplate's volume in optimizer's code.
That said, I'm open for suggestions; `AbstractValue`? `OptValue`?